source/common/memory/stats.cc

```diff
@@ -107,6 +107,20 @@ void Stats::dumpStatsToLog() {
 #endif
 }
 
+std::string Stats::dumpStats() {
+#if defined(TCMALLOC)
+  return tcmalloc::MallocExtension::GetStats();
+#elif defined(GPERFTOOLS_TCMALLOC)
+  constexpr int buffer_size = 100000;
+  std::string buffer(buffer_size, '\0');
+  MallocExtension::instance()->GetStats(buffer.data(), buffer_size);
+  buffer.resize(strlen(buffer.c_str()));
+  return buffer;
+#else
+  return "";
```

Comment:
In this case, I think the admin endpoint should return 501 (Not implemented) and some text saying envoy wasn't built with tcmalloc.